src/controllers/institution-controller.js

```diff
@@ -0,0 +1,177 @@
+import * as institutionService from '../services/institution-service.js'
+import { models } from '../models/init-models.js'
+import { Sequelize } from 'sequelize'
+import sequelizeConn from '../util/db.js'
+
+export async function fetchProjectInstitutions(req, res) {
+  try {
+    const institutions = await institutionService.fetchInstitutions(
+      req.params.projectId
+    )
+    res.status(200).json({ institutions })
+  } catch (e) {
+    console.error('Problem while getting project institutions (controller).', e)
+    res
+      .status(500)
+      .json({ message: 'Error while getting project institutions.' })
+  }
+}
+export async function addInstitutionToProject(req, res) {
+  const projectId = req.params.projectId
+  const institutionId = req.body.institutionId
+
+  try {
+    const projectInstitution = models.InstitutionsXProject.build({
+      project_id: projectId,
+      institution_id: institutionId,
+    })
+
+    const transaction = await sequelizeConn.transaction()
+    await projectInstitution.save({
+      transaction: transaction,
+      user: req.user,
+    })
+
+    await transaction.commit()
+    const institution = await models.Institution.findByPk(institutionId)
+    res.status(200).json({ institution })
+  } catch (e) {
+    console.error(e)
+    res.status(500).json({ message: 'could not assign the two' })
+  }
+}
+export async function createInstitution(req, res) {
+  const projectId = req.params.projectId
+  const name = req.body.name
+  const date = new Date()
+
+  const month = date.getMonth() + 1
+  const day = date.getDate()
+  const year = date.getFullYear()
+
+  const dateFormat = parseInt(`${month}${day}${year}`)
+
+  const institution = await models.Institution.findOne({
+    where: { name: name },
+  })
+
+  if (institution != null) {
+    console.error(
+      'Can not build an institution that already exists: ',
+      institution.name
+    )
+    return
+  }
```

Comment:
This condition will never complete the request and the client will just hang/timeout.

But rather than return with an error, let's just skip creating the Institution and then add the institution to the project.